File: 10/10.cc
#include <iostream>

// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.
// Find the sum of all the primes below two million.

bool isPrime(long possiblePrime) {
    long max = possiblePrime / 2;

    if(possiblePrime < 2)
    {
        return false;
    }

    if(possiblePrime < 3)
    {
        return true;
    }

    for( long i = 2; i <= max; i++ ) {
        if( 0 == possiblePrime % i ) {
            return false;
        }
    }

    return true;
}

int main()
{
    long max = 2000000;
    double sum = 0;

    for(long i = 0; i <= max ; i++)
    {
        if(isPrime(i)) {
            // std::cout
            //     << i
            //     << " is prime "
            //     << std::endl;
            sum = sum+i;
        }
    }


    std::cout
        << " the sum of the primes less than "
        << max
        << " is "
        << std::fixed
        << sum
        << std::endl;

    return 0;
}

File: 10/10.py
import sys
import time
import math

# The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

# Find the sum of all the primes below two million.

start = time.time()

def isprime( number ):
    "is number a prime number"
    if(number < 2):
        return False;

    if(number < 4):
        return True;

    for i in range(2, int(math.sqrt(number)) + 1):
        if number % i == 0:
            return False
    return True

runningsum = 0
maxtocheck = 2000000

for i in range(0, maxtocheck):
    if isprime(i):
        runningsum += i
        
print "The sum of primes below " + str(maxtocheck) + " is " + str(runningsum)

end = time.time()
sys.exit("Elapsed: "+str(end - start))

#The sum of primes below 2000000 is 142913828922
#Elapsed: 22.0553779602
